testng-core/src/test/java/test/thread/issue188/Issue188TestSample.java

```diff
@@ -37,7 +37,7 @@ public void anotherSampleTest() {
 
   private void sleepSilently() {
     try {
-      TimeUnit.MILLISECONDS.sleep(500 * random.nextInt(10));
+      TimeUnit.MILLISECONDS.sleep(5000 * random.nextInt(5));
```

Comment:
Previously the sleep was "0..5 seconds", and now the timeout is "0..25 seconds". What is the reason for raising the upper bound only?

Reply:
I am just increasing the sleep value so that the sleep happens. I noticed that this test also fails sporadically when the stress parameters are used.

Reply:
Sorry but I don't understand what is fixed here.

Reply:
Julien - Neither do I to be honest. These failures are very hard for me to reproduce. I am hoping that a bump up in the timeout value may perhaps fix them when running in LCM/GCM mode. Until theres a reliable way to reproduce the issue, I could ONLY do trial and error.

Reply:
Incase you have any ideas on how to reproduce the problem please let me know. Alternatively we can merge this PR and see if it helps.

Reply:
> I am just increasing the sleep value so that the sleep happens

@krmahadevan , `nextInt` can return `0`, and in those cases the `sleep` would be `0`.
If you want to increase the sleep here, you should ensure the code does not generate `0` sleeps.

Reply:
> Julien - Neither do I to be honest. These failures are very hard for me to reproduce. I am hoping that a bump up in the timeout value may perhaps fix them when running in LCM/GCM mode. Until theres a reliable way to reproduce the issue, I could ONLY do trial and error. 

Why not having both test cases? 2x more probabilities to produce the issue.

Reply:
@juherr I am not sure how that is going to help in reproducing the problem. Having a smaller thread sleep is perhaps not giving enough time for the system to behave as intended when those flags are being passed. I am not sure as to what exactly is getting changed when those flags are being passed. From a functional stand point the code seems to work as re-iterated by the passing test. But under the conditions enforced by LCM/GCM it becomes flaky. Rather than merely throwing more tests and hoping that the issue would show up, I would like to be able to reproduce the problem so that it can be fixed. 

@vlsi - Ok. Let me try to fix that by altering the test such that we atleast wait for 5 seconds.

Reply:
Since the intent was to prolong the delay, I chose the path of ensuring that the randomness always returns a non zero value and have pushed the changes.  Please check @juherr @vlsi